app/src/main/java/io/homeassistant/companion/android/sensors/SensorManager.kt

```diff
@@ -36,19 +37,20 @@ interface SensorManager {
         var sensor = sensorDao.get(sensorId)
         val permission = checkPermission(context)
 
-        // If we haven't created the entity yet do so and default to enabled
-        if (sensor == null) {
+        // If we haven't created the entity yet do so and default to enabled if required
+        if (sensor == null && enabledByDefault) {
             sensor = Sensor(sensorId, permission, false, "")
```

Comment:
```suggestion
        if (sensor == null) {
            sensor = Sensor(sensorId, permission && enabledByDefault, false, "")
```

Reply:
By making this change the bellow changes won't be needed.  It will ensure that sensor is never null going forward.

Reply:
oh much better :) thank you